processor/tailsamplingprocessor/processor.go

```diff
@@ -336,13 +327,15 @@ func (tsp *tailSamplingSpanProcessor) processTraces(resourceSpans ptrace.Resourc
 		for i := 0; i < lenPolicies; i++ {
 			initialDecisions[i] = sampling.Pending
 		}
-		initialTraceData := &sampling.TraceData{
-			Decisions:   initialDecisions,
-			ArrivalTime: time.Now(),
-			SpanCount:   atomic.NewInt64(lenSpans),
+		d, loaded := tsp.idToTrace.Load(id)
+		if !loaded {
+			d, loaded = tsp.idToTrace.LoadOrStore(id, &sampling.TraceData{
```

Comment:
Why calling Load+LoadOrStore now?

Reply:
Because does not make any sense to allocate TraceData if already there.